### DISCUSSION TITLE
Request: 閲覧権限設定時に複数グループを選択したい

### DISCUSSION THREAD

**Discussion by nakashimaki** [growilabs/growi#7429] [Ideas / Requests]
## 概要

ページに対する閲覧権限として特定の1グループを設定可能ですが、複数のグループを選択できるようになってほしいです。

## 背景

特定チームのみに見せたいページのアクセス権設定として現状のグループ設定機能を重宝しています。  
しかし、複数のチームにまたがる内容にアクセス権を設定したい場合に打てる手が「両方のメンバーを含むグループを作成する」しかないため、適切な設定を行うためには手間がかかってしまいます。  
結果としてGROWI以外の情報共有手段に流れてしまうケースがあります。

複数チームにまたがって幅広くGROWI使用する際の使い勝手向上につながると思いますのでご検討いただけますと幸いです！

## 他の解決手段の案

グループの子として複数のグループを設定できるようにすることによっても課題が解決できると考えています。

例：

* チームA: 佐藤、鈴木、高橋が所属するグループ
* チームB: 田中、伊藤が所属するグループ

* プロジェクトC: チームA、チームBが所属するグループ
   * 佐藤、鈴木、高橋、田中、伊藤が所属するグループとして機能する

**Comment by misafortune** [comment]
同意します。
グループ複数選択が出来るようになると、かなり楽になりますね！
現状、複雑にネスト化されたグループツリーを作成しないと対応出来ないでいます。

**Comment by yuki-takei** [comment]
ご意見ありがとうございます。

別の Discussion ですが、LDAP 由来のグループとの同期機能は現在開発中です。(https://github.com/weseek/growi/discussions/7446)
その他のグループ関連施策はこちらが終わり次第改めて着手検討したいと思っています。


> ページに対する閲覧権限として特定の1グループを設定可能ですが、複数のグループを選択できるようになってほしいです。

そうですね、本命はこちらだと思います。


> グループの子として複数のグループを設定できるようにすることによっても課題が解決できると考えています。

こちらは現状の機能で可能です。

チームC を親グループとし、チームA、チームB を C の子グループとして定義することで、AまたはBに加えられたユーザーは自動的にCにも加えられます。

**Comment by yuki-takei** [comment]
v6.3.2 で複数グループの設定 および LDAP/Keycloak からのグループ同期が実装されました。お試しください。
https://docs.growi.org/ja/admin-guide/upgrading/63x.html
